src/jit/codegencommon.cpp

```diff
@@ -647,7 +647,7 @@ regMaskTP Compiler::compHelperCallKillSet(CorInfoHelpFunc helper)
 #if defined(_TARGET_AMD64_)
             return RBM_RSI | RBM_RDI | RBM_CALLEE_TRASH;
 #elif defined(_TARGET_ARM64_)
-            return RBM_CALLEE_TRASH_NOGC;
+            return RBM_R12 | RBM_R15;
```

Comment:
OK We do want the code sequence to look like you showed above:
As long as you have verified that we generate the correct sequence and that after the call R13 and R14 are alive we should be good.